src/purchase-client.ts

```diff
@@ -0,0 +1,203 @@
+import { MuxySDK } from ".";
+import { CurrentEnvironment } from './util';
+import Config from './config';
+
+import { TwitchBitsProduct, TwitchBitsTransaction } from './twitch';
+
+// Muxy dev variable
+declare global {
+  interface Window {
+    MEDKIT_PURCHASABLE_ITEMS: Product[] | PromiseLike<Product[]>;
+  }
+}
+
+export enum PurchaseClientType {
+  Dev,
+  Server,
+  Twitch,
+  Unknown
+}
+
+export interface PurchaseClient {
+  getProducts(): Promise<Product[]>;
+  purchase(identifier: string): void;
+  onUserPurchase(callback: (transaction: Transaction) => void): void;
+}
+
+export interface Cost {
+  amount: number;
+  type: string;
+}
+
+export interface Product {
+  sku: string;
+  displayName: string;
+  cost: Cost;
+}
+
+export interface Transaction {
+  transactionId: string;
+  product: Product;
+  userId: string;
+  displayName: string;
+  initiator: any;
+  transactionReceipt: string;
+}
+
+// TwitchPurchaseClient implements the basic 'purchase client' interface.
+// This is used by SDK to provide low-level access to twitch bits transactions.
+class TwitchPurchaseClient implements PurchaseClient {
+  /**
+   * purchase will start the twitch bits transaction.
+   *
+   * @since 2.4.0
+   *
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   * 
+   * @param identifier the twitch sku of the item to be purchased.
+   *
+   * @example
+   * client.purchase("XXSKU000");
+   */
+  public purchase(identifier: string) {
+    if (!window.Twitch.ext.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+
+    window.Twitch?.ext?.bits.useBits(identifier);
+  }
+ 
+  /**
+   * Returns a list of Twitch Bits Products for the current channel.
+   * 
+   * @async
+   * @since 2.4.0
+   * 
+   * @throws {TwitchHelperError} Will throw an error if the Twitch Helper didn't load.
+   *
+   * @return {Promise<[]TwitchBitsProduct>} Resolves with an array of {@link TwitchBitsProduct}
+   * objects for each available sku.
+   *
+   * @example
+   * const products = client.getProducts();
+   */
+  public getProducts(): Promise<TwitchBitsProduct[]> {
+    const target = window.Twitch?.ext?.bits;
+    if (!window.Twitch.ext.bits) {
+      throw new Error('Twitch helper was not loaded.');
+    }
+    
+    return target.getProducts();
```

Comment:
You probably don't need the assignment to `target` unless you also want to change your check to `if (!target)`